### PR TITLE
Implements all kind of resource modifiers.

### PR DESCRIPTION
**Pull request by GraionDilach**
~~Depends on #15282.~~
Related to #15220.

### CODE REVIEW

**Comment by GraionDilach** [comment]
Pinging @MustaphaTR for review (or atleast a confirmation that it takes care of his GenAlpha usecase).

**Comment by MustaphaTR** [comment]
Something is wrong with the timer for ExternalResourceProvider's ticks. First bale since existance of modifier actor gets a seperate tick. Here are MSLOs with 25% modifier.

resim

File: OpenRA.Mods.Common/Traits/ExternalResourceMultiplier.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+		public override object Create(ActorInitializer init) { return new ExternalResourceMultiplier(init.Self, this); }
+	}
+
+	public class ExternalResourceMultiplier : ConditionalTrait<ExternalResourceMultiplierInfo>, ITick, INotifyOwnerChanged
+	{
+		readonly ExternalResourceMultiplierInfo info;
+
+		PlayerResources playerResources;
+		int currentDisplayTick = 0;
```
[the remark this answers is not in the document]
Comment (GraionDilach) [reply OpenRA.Mods.Common/Traits/ExternalResourceMultiplier.cs]: Ah, yes, you're indeed correct, just tested.

**Comment by GraionDilach** [comment]
Updated.

**Comment by pchote** [comment]
Moving to the Future milestone so we don't lose this.

**Comment by MustaphaTR** [comment]
This looks OK to get as is, i dunno if i should open a seperate PR.

**Comment by GraionDilach** [comment]
Rebased.

**Pull request closed without merging**